client/src/components/settings/Username.js

```diff
@@ -77,7 +81,8 @@ class UsernameSettings extends Component {
       /* eslint-disable-next-line react/no-did-update-set-state */
       return this.setState({
         isFormPristine: username === formValue,
-        submitClicked: false
+        submitClicked: false,
+        isUserNew: tempUserRegex.test(formValue)
```

Comment:
```suggestion
        isUserNew: tempUserRegex.test(username)
```
While I'm not sure *why* the `formValue` would be different from the `username` at this point, it is the `username` that matters.